### PR TITLE
NXP-29414: add a release pipeline

### PR DESCRIPTION
**Pull request by troger**

### CODE REVIEW

File: ci/Jenkinsfiles/release.groovy [hunk 1]
```diff
@@ -0,0 +1,215 @@
+/*
+ * (C) Copyright 2019 Nuxeo (http://nuxeo.com/) and others.
```
Comment (ataillefer) [inline ci/Jenkinsfiles/release.groovy]:
```suggestion
 * (C) Copyright 2020 Nuxeo (http://nuxeo.com/) and others.
```

File: ci/Jenkinsfiles/release.groovy [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+          }
+        }
+      }
+    }
+
+    stage('Bump master branch') {
+      steps {
+        container('maven') {
+          script {
+            def releaseVersion = readMavenPom().getVersion()
```
Comment (ataillefer) [inline ci/Jenkinsfiles/release.groovy:170]: Maybe `git checkout release` just before to be explicit and know from which branch the release version comes from?

File: versions-rules.xml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+<ruleset comparisonMethod="maven"
+         xmlns="http://mojo.codehaus.org/versions-maven-plugin/rule/2.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://mojo.codehaus.org/versions-maven-plugin/rule/2.0.0 http://mojo.codehaus.org/versions-maven-plugin/xsd/rule-2.0.0.xsd">
+</ruleset>
```
Comment (ataillefer) [inline versions-rules.xml:4]: Is this file needed?

File: ci/Jenkinsfiles/release.groovy [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+      steps {
+        container('maven') {
+          echo """
+          ----------------------------------------
+          Upload Nuxeo Packages to ${CONNECT_PROD_URL}
+          ----------------------------------------
+          """
+          withCredentials([usernameColonPassword(credentialsId: 'connect-prod', variable: 'CONNECT_PASS')]) {
+            sh """
+              PACKAGES_TO_UPLOAD="packages/nuxeo-*-package/target/nuxeo-*-package-*.zip"
```
Comment (ataillefer) [inline ci/Jenkinsfiles/release.groovy]: Might not work for the `nuxeo-jsf-ui` package itself: `packages/nuxeo-marketplace-jsf-ui/target/nuxeo-marketplace-jsf-ui-11.2-SNAPSHOT.zip`.

Comment (troger) [reply ci/Jenkinsfiles/release.groovy]: Good catch!

File: versions-rules.xml (hunk 3 above)
Comment (troger) [reply versions-rules.xml:4]: Yes, for the Maven versions plugin, otherwise `versions:update-parent` complains that there is no version rules.

File: ci/Jenkinsfiles/release.groovy (hunk 2 above)
Comment (troger) [reply ci/Jenkinsfiles/release.groovy:170]: Added some comments too.

File: ci/Jenkinsfiles/release.groovy [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+ *     Thomas Roger <troger@nuxeo.com>
+ */
+
+properties([
+  [$class: 'GithubProjectProperty', projectUrlStr: 'https://github.com/nuxeo/nuxeo-jsf-ui/'],
+  [$class: 'BuildDiscarderProperty', strategy: [$class: 'LogRotator', daysToKeepStr: '60', numToKeepStr: '60', artifactNumToKeepStr: '5']],
+  disableConcurrentBuilds(),
+])
+
+void setGitHubBuildStatus(String context, String message, String state) {
```
Comment (ataillefer) [inline ci/Jenkinsfiles/release.groovy]: unused

File: ci/Jenkinsfiles/release.groovy [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+            sh "mvn ${MAVEN_ARGS} install"
+
+            if (env.DRY_RUN != "true") {
+              sh """
+                jx step git credentials
+                git config credential.helper store
+
+                git push origin v${releaseVersion}
+              """
+              currentBuild.description = "Release ${releaseVersion}"
```
Comment (ataillefer) [inline ci/Jenkinsfiles/release.groovy:112]: Do we want it there or as a post success of all stages as in the [nuxeo release pipeline](https://github.com/nuxeo/nuxeo/blob/master/ci/Jenkinsfiles/release.groovy#L274)?

Comment (troger) [reply ci/Jenkinsfiles/release.groovy:112]: Make sense to keep it here if the Git tag was pushed, and that avoid us to refetch the `releaseVersion` from the post success... but I'm OK with both 🤷

**Review by ataillefer — DISMISSED** [review]

**Review by ataillefer — DISMISSED** [review]

**Review by kevinleturc — APPROVED** [review]

**Review by ataillefer — APPROVED** [review]
